spec/index.bs

```diff
@@ -3726,40 +3726,26 @@ GPURenderPipeline includes GPUPipelineBase;
 
 <script type=idl>
 dictionary GPURenderPipelineDescriptor : GPUPipelineDescriptorBase {
-    required GPUProgrammableStageDescriptor vertexStage;
-    GPUProgrammableStageDescriptor fragmentStage;
-
-    required GPUPrimitiveTopology primitiveTopology;
-    GPURasterizationStateDescriptor rasterizationState = {};
-    required sequence<GPUColorStateDescriptor> colorStates;
-    GPUDepthStencilStateDescriptor depthStencilState;
-    GPUVertexStateDescriptor vertexState = {};
-
-    GPUSize32 sampleCount = 1;
-    GPUSampleMask sampleMask = 0xFFFFFFFF;
-    boolean alphaToCoverageEnabled = false;
+    GPUVertexInputState vertexInput = {};
+    required GPUProgrammableStage vertexStage;
+    GPUPrimitiveAssemblerState primitiveAssembler = {};
+    GPURasterizerState rasterizer = {};
+    GPUProgrammableStage fragmentStage;
+    GPUDepthStencilState depthStencil;
+    required GPUOutputMergerState outputMerger;
```

Comment:
IMHO `outputMerger` is a bit weird, because looking at D3D12 (from which the terminology comes) OM is only used for `OMSetStencilRef` `OMSetBlendColor` `OMSetDepthBounds` and `OMSetRenderTargets`. The `D3D12_GRAPHICS_PIPELINE_STATE_DESC` has `AlphaToCoverage` in the BLEND_DESC.

The extra indirection with the `outputMerge.colorOutputs` is a bit big for something that's so useful. Maybe we could keep the `colorOutputs` directly in `GPURenderPipelineDescriptor` and put all the multisampling state together in a `multisample` sub-descriptor. It doesn't necessarily follow the ordering of the graphics pipeline, but puts all the information you care about for multisampling in the same place.

Reply:
that would basically sabotage #936 :(

Reply:
I like the current revision (.colorWrite.outputs), but if we wanted to remove the indirection, we could put alphaToCoverageEnabled *inside* the GPUColorOutputState and allow it to be set only on index 0.

Reply:
Right, that's an interesting option. I'd prefer to not go this way, since we may want to add more fields in the write output structure, e.g. with extensions, and we don't want to be forced to put them elsewhere.